lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -9,6 +9,26 @@ class AwsCompileStreamEvents {
     this.provider = this.serverless.getProvider('aws');
 
     this.hooks = {
+      'initialize': () => {
+        if (
+          this.serverless.service.provider.name === 'aws' &&
+          Object.values(this.serverless.service.functions).some(({ events }) =>
+            events.some(({ stream: eventObject }) => {
+              const consumer = eventObject && eventObject.consumer;
+              const usesServiceSpecificNamingMode =
+                this.serverless.service.provider.kinesis &&
+                this.serverless.service.provider.kinesis.consumerNamingMode &&
+                this.serverless.service.provider.kinesis.consumerNamingMode === 'serviceSpecific';
+              return consumer && !usesServiceSpecificNamingMode;
+            })
+          )
+        ) {
+          this.serverless._logDeprecation(
+            'KINESIS_CONSUMER_NAME_CONTAINING_SERVICE',
+            'Consumer name for Kinesis streams will be different in v3. The new naming will cause re-deployment which may lead to delay in Kinesis stream data processing.'
```

Comment:
Let's be a little bit more clear here and follow similar messaging to already existing deprecation notices. 

Maybe something along those lines? (It's definitely not a final version) 

```
Starting with v3.0.0, generated consumer name for Kinesis streams will use a new naming scheme that will include service and stage to ensure uniqueness across different services and/or stages. In order to hide the deprecation and ensure seamless upgrade, please set the property ...
```